### ISSUE TITLE
What is this supposed to mean.

### ISSUE THREAD

**Issue by rdebath** [koalaman/shellcheck#947]
This seems to be a complete false positive. But I'm not sure because the message is so confused that I can't work out what it's supposed to be looking for. The errors do not have pages in the wiki.

The script works and does the same thing on every Bourne shell I can find: bash, dash, ash, posh, zsh, ksh, ksh88, v7sh.

Yup, that everything all the way back to the Unix V7 version.

#### For bugs
- Rule Id (if any, e.g. SC1000): SC1041, SC1042, SC1072
- My shellcheck version (`shellcheck --version` or "online"): Online and version: 0.4.4
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
cat <<'#!'
Hello World
#!
echo Done
```

#### Here's what shellcheck currently says:
/tmp/p2.sh|3 col 1 error| Close matches include '#!' (!= '#!'). [SC1041]
/tmp/p2.sh|3 col 1 error| Found '#!' further down, but not entirely by itself. [SC1042]
/tmp/p2.sh|6 col 1 error| Fix any mentioned problems and try again. [SC1072]

**Comment by koalaman** [comment]
Very confusing. This is definitely a parsing bug. 

To give helpful messages on wrong indenting of the end token, ShellCheck skipped spaces before the token. Unfortunately, the particular function that skipped spaces also skipped comments. Such as `#!`. 

This has now been fixed. There's still a similar problem with tokens containing leading spaces, but that's for another commit.

**Comment by koalaman** [comment]
Oh, and as a workaround until this fix makes it into a release, you can use any token that doesn't start with `#`. `end` and `eof` are common choices.

**Comment by rdebath** [comment]
Heh, ah no, "end" and "eof" wouldn't work because they're not valid in the awk script:

This is a more complete example, it's actually an awk script ... well, sorta.
``` bash
#!/bin/sh
# vim: set filetype=awk.sh:
true ,/^; #/&&0{$1="                                                     \
                                                                        #\
LANG=C; export LANG                                                     #\
[ "$1" = all ] && { FULLLIST=1 ; shift; }                               #\
[ "$1" = me ] && { FULLLIST=1 ; NOSECTION=1 ; shift; }                  #\
                                                                        #\
OUTFILE="$1"                                                            #\
case "$OUTFILE" in                                                      #\
*.txt )     FMT=wiki ;;                                                 #\
*.html )    FMT=html ;;                                                 #\
"" )        OUTFILE=/dev/stdout; FMT=html ;;                            #\
* )         FMT=html ;;                                                 #\
esac                                                                    #\
                                                                        #\
AWK=awk ; MAWK=$(which mawk)                                            #\
[ -f "$MAWK" ] && AWK=mawk                                              #\
                                                                        #\
# Use sed not cat to append an EOL to each file if needed.              #\
find Testrun/Out Testrun/sav -name '*.log' -print | sort |              #\
xargs sed '' | cut -c1-200 |                                            #\
$AWK -v Format="$FMT" -v fulllist=$FULLLIST -v nosection=$NOSECTION      \
    -f "$0" -f etc/PrgURL > "$OUTFILE" ;exit;:<<'#!'                  #";}

BEGIN {

    lasttest = ""
    runtime = ""
    gothi = 0
    segfault = 0
    overlimit = 0
    gotgoodtime = 0
    lastinterim = 0

    hash["Hello"] = "8ddd8be4b179a529afa5f2ffae4b9858";
    hash["666"] = "971658bc2f5bdee5660844a83b5bf0a2";

### etc etc 


}

#!
```

I guess Syntastic started shellcheck because of the `#!/` line but it shouldn't have ignored the filetype line.
